include/lsst/afw/image/lsstGil.h

```diff
@@ -40,20 +40,8 @@
 #include "boost/mpl/bool.hpp"
 #include "boost/mpl/if.hpp"
 
-//#define BOOST_GIL_USE_CONCEPT_CHECK 1
 
-#if defined(__ICC)
-#pragma warning(push)
-#pragma warning(disable : 68)
-#pragma warning(disable : 304)
-#endif
```

Comment:
We don't officially support icc, nor do we have a good way to test it, but I'd rather not remove this without confirmation that it's unneeded from someone that does use it.

Reply:
I tried ICC 2021 and compilation didn't work at all against the 0.7.0 conda stack.

Reply:
I wouldn't expect it to; conda-forge is explicitly gcc.  An icc user would likely have to provide icc versions of all C++ dependencies.  (Same goes for clang on Linux, which is, interestingly enough, distributed by Linux conda-forge but not usable as is.)  Maybe that's too difficult to even contemplate anymore.

Reply:
My take on this is if icc compilation doesn't work, the question is whether it's us or them that isn't standards compliant.  I'm all for testing more compilers to check our compliance, but not really interested in trying at all to support one we can't easily test if it's because they don't fully support the standard we target.

Reply:
clang on Linux only works if lsst_distrib, ci_*, etc is compiled completely against the coda stack. I would assume pybind11 is the culprit here as compiling single packages with clang/Linux fails at the python test level.  The icc compilation problem looked more fundamental to me as one compiler flag passed to icc was not recognized. icc would probably also need support by the build system.

Reply:
I'll keep the pragmas for now. Slightly beyond the scope of the ticker anyway.

Reply:
I don't think this makes sense to keep, just to suppress a few warnings. The whole boost gil stuff is a mess anyway as it injects lsst code into boost namespace.

Reply:
OK, you've convinced me.  Take it out. :)